crates/bevy_core_pipeline/src/dof/mod.rs

```diff
@@ -0,0 +1,907 @@
+    }
+}
+
+/// The node in the render graph for depth of field.
+#[derive(Default)]
+pub struct DepthOfFieldNode;
+
+/// The layout for the bind group shared among all invocations of the depth of
+/// field shader.
+#[derive(Resource, Clone)]
+pub struct DepthOfFieldGlobalBindGroupLayout {
+    /// The layout.
+    layout: BindGroupLayout,
+    /// The sampler used to sample from the color buffer or buffers.
+    color_texture_sampler: Sampler,
+}
+
+/// The bind group shared among all invocations of the depth of field shader,
+/// regardless of view.
+#[derive(Resource, Default, Deref, DerefMut)]
+pub struct DepthOfFieldGlobalBindGroup(Option<BindGroup>);
+
+#[derive(Component)]
+pub enum DepthOfFieldPipelines {
+    Gaussian {
+        horizontal: CachedRenderPipelineId,
+        vertical: CachedRenderPipelineId,
+    },
+    Bokeh {
+        pass_0: CachedRenderPipelineId,
+        pass_1: CachedRenderPipelineId,
+    },
+}
+
+struct DepthOfFieldPipelineRenderInfo {
+    pass_label: &'static str,
+    view_bind_group_label: &'static str,
+    pipeline: CachedRenderPipelineId,
+    is_dual_input: bool,
+    is_dual_output: bool,
+}
+
+/// The extra texture used as the second render target for the hexagonal bokeh
+/// blur.
+///
+/// This is the same size and format as the main view target texture. It'll only
+/// be present if bokeh is being used.
+#[derive(Component, Deref, DerefMut)]
+pub struct AuxiliaryDepthOfFieldTexture(CachedTexture);
+
+/// Bind group layouts for depth of field specific to a single view.
+#[derive(Component, Clone)]
+pub struct ViewDepthOfFieldBindGroupLayouts {
+    /// The bind group layout for passes that take only one input.
+    single_input: BindGroupLayout,
+
+    /// The bind group layout for the second bokeh pass, which takes two inputs.
+    ///
+    /// This will only be present if bokeh is in use.
+    dual_input: Option<BindGroupLayout>,
+}
+
+/// Information needed to specialize the pipeline corresponding to a pass of the
+/// depth of field shader.
+pub struct DepthOfFieldPipeline {
+    /// The bind group layouts specific to each view.
+    view_bind_group_layouts: ViewDepthOfFieldBindGroupLayouts,
+    /// The bind group layout shared among all invocations of the depth of field
+    /// shader.
+    global_bind_group_layout: BindGroupLayout,
+}
+
+impl ViewNode for DepthOfFieldNode {
+    type ViewQuery = (
+        Read<ViewUniformOffset>,
+        Read<ViewTarget>,
+        Read<ViewDepthTexture>,
+        Read<DepthOfFieldPipelines>,
+        Read<ViewDepthOfFieldBindGroupLayouts>,
+        Read<DynamicUniformIndex<DepthOfFieldUniform>>,
+        Option<Read<AuxiliaryDepthOfFieldTexture>>,
+    );
+
+    fn run<'w>(
+        &self,
+        _: &mut RenderGraphContext,
+        render_context: &mut RenderContext<'w>,
+        (
+            view_uniform_offset,
+            view_target,
+            view_depth_texture,
+            view_pipelines,
```

Comment:
Maybe consider renaming this to view_dof_pipelines. I got confused for a second where the dof pipelines where, I figured it out pretty quickly by just reading the code but this could have made it slightly more straightforward. Not a particularly important thing to do though.